packages/compass-connections/src/hooks/use-connection-repository.ts

```diff
@@ -122,20 +127,36 @@ export function useConnectionRepository(): ConnectionRepository {
         }
       }
 
-      await storage.save({ connectionInfo: infoToSave });
+      if (storage.save) {
+        await storage.save({ connectionInfo: infoToSave });
+      }
```

Comment:
Just fyi, optional chaining syntax works for functions too:

```suggestion
      await storage.save?.({ connectionInfo: infoToSave });
```

(mentioning because it's a bit inconsistent through the code changes here)

Reply:
Ahh yea, true that. I will fix this through.